cmd/medius/images/promote.go

```diff
@@ -0,0 +1,103 @@
+package images
+
+import (
+	"context"
+	"errors"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"kubevirt.io/containerdisks/cmd/medius/common"
+	"kubevirt.io/containerdisks/pkg/api"
+	"kubevirt.io/containerdisks/pkg/repository"
+)
+
+func NewPromoteImagesCommand(options *common.Options) *cobra.Command {
+	options.PromoteImageOptions = common.PromoteImageOptions{
+		TargetRegistry: "quay.io/containerdisks",
+	}
+
+	promoteCmd := &cobra.Command{
+		Use:   "promote",
+		Short: "Promote verified containerdisks from one registry to another registry",
+		Run: func(cmd *cobra.Command, args []string) {
+			results, err := readResultsFile(options.ImagesOptions.ResultsFile)
+			if err != nil {
+				logrus.Fatal(err)
+			}
+
+			resultsChan, err := spawnWorkers(cmd.Context(), options, func(a api.Artifact) (*api.ArtifactResult, error) {
+				r, ok := results[a.Metadata().Describe()]
+				if !ok || r.Stage != StageVerify || r.Err != "" {
+					return nil, nil
+				}
+
+				errString := ""
+				err := promoteArtifact(cmd.Context(), a, r.Tags, options)
+				if err != nil {
+					errString = err.Error()
+				}
+
+				return &api.ArtifactResult{
+					Tags:  r.Tags,
+					Stage: StagePromote,
+					Err:   errString,
+				}, err
+			})
+
+			for result := range resultsChan {
+				results[result.Key] = result.Value
+			}
+
+			if !options.DryRun {
+				if err := writeResultsFile(options.ImagesOptions.ResultsFile, results); err != nil {
+					logrus.Fatal(err)
+				}
+			}
```

Comment:
I think that after writing the result file, we want to fail if we ever saw a `r.Err != ""` and properly explain why we fail.